android/src/main/java/com/optimove/flutter/OptimoveFlutterSdkPlugin.java

```diff
@@ -0,0 +1,344 @@
+package com.optimove.flutter;
+
+import android.app.Activity;
+
+import androidx.annotation.NonNull;
+
+import com.optimove.android.Optimove;
+import com.optimove.android.optimobile.InAppInboxItem;
+import com.optimove.android.optimobile.Optimobile;
+import com.optimove.android.optimobile.OptimoveInApp;
+
+import org.json.JSONException;
+
+import java.lang.ref.WeakReference;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TimeZone;
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.plugin.common.EventChannel;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+
+/** OptimoveFlutterSdkPlugin */
+public class OptimoveFlutterSdkPlugin implements FlutterPlugin, MethodCallHandler, ActivityAware {
+  /// The MethodChannel that will the communication between Flutter and native Android
+  ///
+  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+  /// when the Flutter Engine is detached from the Activity
+  private MethodChannel channel;
+  private EventChannel eventChannel;
+
+  static WeakReference<Activity> currentActivityRef = new WeakReference<>(null);
+  static QueueingEventStreamHandler eventSink = new QueueingEventStreamHandler();
+
+  @Override
+  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
+    channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "optimove_flutter_sdk");
+    channel.setMethodCallHandler(this);
+
+    eventChannel = new EventChannel(flutterPluginBinding.getBinaryMessenger(), "optimove_flutter_sdk_events");
+    eventChannel.setStreamHandler(eventSink);
+  }
+
+  @Override
+  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
+    channel.setMethodCallHandler(null);
+    eventChannel.setStreamHandler(null);
+    eventSink.onCancel(null);
+  }
+
+  @Override
+  public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
+    currentActivityRef = new WeakReference<>(binding.getActivity());
+  }
+
+  @Override
+  public void onDetachedFromActivityForConfigChanges() {
+    currentActivityRef = new WeakReference<>(null);
+  }
+
+  @Override
+  public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding binding) {
+    currentActivityRef = new WeakReference<>(null);
```

Comment:
Should this be `null` in the ref?

Reply:
(ok perhaps an existing copy/paste bug: https://github.com/Kumulos/KumulosSdkFlutter/blob/f03f6d848d629a9b93827223a2a01353cd9b0469/android/src/main/java/com/kumulos/flutter/kumulos_sdk_flutter/KumulosSdkFlutterPlugin.java#L105-L107)

Reply:
Fixed